packages/venia-concept/webpack.config.js

```diff
@@ -56,7 +58,7 @@ module.exports = async env => {
              */
             STORE_NAME: JSON.stringify('Venia')
         }),
-        new MediaBackendUrlFetcherPlugin(),
+        new AsyncWebpackPlugin(getMediaURL),
```

Comment:
Since the export of this module is already an `async` function, another approach I've seen is to just do the async work before returning your webpack config object return value:

```js
module.exports = async env => {
    const mediaUrl = await getMediaURL();

    const config = await configureWebpack({ ... });

    return config;
};
```

Reply:
Doing so will move the async logic from the plugin to the exports function which might become clumsy when we start adding more async functionalities like fetching store name for instance in the future. Also, this will be defeating the purpose of the plugin.

I have been reading on plugins and apparently we can use webpack plugins outside `plugins` object, since they are just JS classes at the end of the day. Leaving logic for async functionality inside the plugin gives us more grip over the usage as well.

This is a long shot but if there is a new and more optimized way to do async stuff in JS in the future, having such things in 1 place will make it easy to perform modifications.

Reply:
> this will be defeating the purpose of the plugin.

Yes 😄 
I'd rather we create and maintain a bunch of utility functions than webpack plugins though.

> Doing so will move the async logic from the plugin to the exports function

Looking at this line for example:

```js
const mediaUrl = await getMediaURL();
```

We move the async logic to `getMediaURL`, with a simple call to that function in the exports function.

> which might become clumsy

It's good to keep an eye on this 👍 . I think straightforward utility functions are less clumsy than webpack plugins in the end though.

Reply:
My only problem with performing such async stuff as part of configureWebpack is code maintainability.

```es6
module.exports = async env => {
    const mediaUrl = await getMediaURL();
    const storeName = await getStoreName();
    const timeZone = await getTimeZone();
    ....
    const config = await configureWebpack({
      // use mediaUrl somewhere in here
    });

    return config;
};
```